### PR TITLE
[DCOS-59137] Introduce the dcos-sdk-service-diagnostics component version managment.

### PR DESCRIPTION
**Pull request by VladimirKravtsov**
This PR addresses the issue with image tag hardcode in `create_service_diagnostics_bundle.sh` script.
This PR depends on #15, so this one is temporary points to DCOS-42309 branch rather than master.

### CODE REVIEW

File: python/create_service_diagnostics_bundle.sh [hunk 1]
```diff
@@ -5,11 +5,10 @@ set -eu -o pipefail
 readonly SCRIPT_DIRECTORY="$(cd "$(dirname "${BASH_SOURCE[0]}")" >/dev/null && pwd)"
 readonly DCOS_SERVICE_DIAGNOSTICS_SCRIPT_PATH="dcos-sdk-service-diagnostics/python"
 
-#ToDo: the image version should be externally managed outside this file.
-readonly VERSION='v0.4.0'
+readonly BUILD_IMAGE_VERSION=${BUILD_IMAGE_VERSION:-$(<VERSION)}
```
Comment (porridge) [inline python/create_service_diagnostics_bundle.sh]: Can you please explain how is this going to work on an end-user's machine?

Comment (VladimirKravtsov) [reply python/create_service_diagnostics_bundle.sh]: Right question and 
@mpereira has the same right question. I think, we can replace ${BUILD_IMAGE_VERSION:-$(<VERSION)}` part with concrete version before upload the script to S3 for GA. Making it `readonly BUILD_IMAGE_VERSION=v0.4.1`. I will update CONTRIBUTING.md appropriately.

Comment (mpereira) [reply python/create_service_diagnostics_bundle.sh]: If that's the approach taken, let's keep the environment variable name as just `VERSION`.

Comment (VladimirKravtsov) [reply python/create_service_diagnostics_bundle.sh]: `VERSION` is too ambiguous name, on my taste. Is it version of the script, or only docker image version. BUILD_IMAGE_VERSION is more concrete.

Comment (mpereira) [reply python/create_service_diagnostics_bundle.sh]: From the user perspective, there should only be one version: the DC/OS Service Diagnostics version. IMO, the current _implementation detail_ of having a versioned script that runs a versioned Docker image shouldn't be made apparent in the user-facing software (the script).

Also, from a user perspective, the script should never have a different version than a Docker image. They're one: the DC/OS Service Diagnostics version.

Comment (VladimirKravtsov) [reply python/create_service_diagnostics_bundle.sh]: PTAL. Also updated instruction in CONTRIBUTING.md.

**Comment by mpereira** [comment]
I pushed a commit making the version-related release steps be automated instead of instructions in prose: https://github.com/mesosphere/dcos-sdk-service-diagnostics/pull/16/commits/fd82d8586f2bc3ac8592cfd78636e1e14cc2d5bf.

**Pull request merged**